### PR TITLE
Moved movies and actors to recyclerview lists from hardcode xml

### PR DESCRIPTION
**Pull request by frANTicBee666**

### CODE REVIEW

File: app/src/main/java/com/android/academy/fundamentals/homework/features/moviedetails/MovieDetailsFragment.kt [hunk 1]
```diff
@@ -29,10 +39,25 @@ class MovieDetailsFragment : Fragment() {
         return inflater.inflate(R.layout.fragment_movie_details, container, false)
     }
 
-    override fun onStart() {
-        super.onStart()
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val movieData = arguments?.getSerializable(PARAM_MOVIE_DATA) as? MovieData ?: return
```
Comment (emartynov) [inline app/src/main/java/com/android/academy/fundamentals/homework/features/moviedetails/MovieDetailsFragment.kt:45]: Question - what is the better - do it here or in the view creation?

File: app/src/main/java/com/android/academy/fundamentals/homework/features/moviedetails/MovieDetailsFragment.kt [hunk 2]
```diff
@@ -43,11 +68,51 @@ class MovieDetailsFragment : Fragment() {
         super.onDetach()
     }
 
+    private fun updateMovieDetailsInfo(movieData: MovieData) {
```
Comment (emartynov) [inline app/src/main/java/com/android/academy/fundamentals/homework/features/moviedetails/MovieDetailsFragment.kt:71]: If you pass view as function param, then you don't need all `?` here

File: app/src/main/java/com/android/academy/fundamentals/homework/features/movies/MoviesListAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                R.color.color_white
+            }
+            ImageViewCompat.setImageTintList(
+                likeImage, ColorStateList.valueOf(
+                    ContextCompat.getColor(likeImage.context, likeColor)
+                )
+            )
+
+            //set stars tint
+            starsImages.forEachIndexed { index, imageView ->
```
Comment (emartynov) [inline app/src/main/java/com/android/academy/fundamentals/homework/features/movies/MoviesListAdapter.kt:72]: A small duplication - I would probably extract a utility method and use it everywhere

File: app/src/main/java/com/android/academy/fundamentals/homework/model/DataGenerator.kt [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            2,
+            16,
+            "Tenet",
+            "Action, Sci-Fi, Thriller",
+            "97 min",
+            98,
+            true,
+            5,
+            R.drawable.img_movies_item_header_tenet,
+            R.drawable.img_movie_details_header,
```
Comment (emartynov) [inline app/src/main/java/com/android/academy/fundamentals/homework/model/DataGenerator.kt:37]: Oh, I missed it in Data class why it is part of data? It looks for me like implementation details about how it is going to be shown.

File: app/src/main/java/com/android/academy/fundamentals/homework/model/MovieData.kt [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    val id: Int,
+    val pgAge: Int,
+    val title: String,
+    val genre: String,
+    val runningTime: String,
+    val reviewCount: Int,
+    val isLiked: Boolean,
+    val rating: Int,
+    @DrawableRes val imageRes: Int,
+    @DrawableRes val detailImageRes: Int,
```
Comment (emartynov) [inline app/src/main/java/com/android/academy/fundamentals/homework/model/MovieData.kt:16]: Is it just mask on top?

**Review by emartynov — COMMENTED** [review]
Would be great to have PR details with screenshots. As for me, XML is always hard to review.

Bunch of minors comments and rest LGTM

**Comment by frANTicBee666** [comment]
homework result. 

Screen Recording 2020-12-13 at 07 11 34 (1)

File: app/src/main/java/com/android/academy/fundamentals/homework/features/moviedetails/MovieDetailsFragment.kt (hunk 1 above)
Comment (frANTicBee666) [reply app/src/main/java/com/android/academy/fundamentals/homework/features/moviedetails/MovieDetailsFragment.kt:45]: We had listeners settings in onStart on 2nd HW, i think onViewCreated is better for this 
BTW, i used to use synthetics, so it did not work propertly in onCreateView

also trying to pass layout name as Fragment constructor parameter helps me to remove onCreateView at all)

File: app/src/main/java/com/android/academy/fundamentals/homework/model/DataGenerator.kt (hunk 4 above)
Comment (frANTicBee666) [reply app/src/main/java/com/android/academy/fundamentals/homework/model/DataGenerator.kt:37]: there is another image here, not from movies list

File: app/src/main/java/com/android/academy/fundamentals/homework/model/MovieData.kt (hunk 5 above)
Comment (frANTicBee666) [reply app/src/main/java/com/android/academy/fundamentals/homework/model/MovieData.kt:16]: no, this is image for movie details

File: app/src/main/java/com/android/academy/fundamentals/homework/model/DataGenerator.kt (hunk 4 above)
Comment (emartynov) [reply app/src/main/java/com/android/academy/fundamentals/homework/model/DataGenerator.kt:37]: Yeah but it is the same for all movies

File: app/src/main/java/com/android/academy/fundamentals/homework/model/MovieData.kt (hunk 5 above)
Comment (emartynov) [reply app/src/main/java/com/android/academy/fundamentals/homework/model/MovieData.kt:16]: It is the same for all movies

File: app/src/main/java/com/android/academy/fundamentals/homework/features/moviedetails/MovieDetailsFragment.kt (hunk 1 above)
Comment (emartynov) [reply app/src/main/java/com/android/academy/fundamentals/homework/features/moviedetails/MovieDetailsFragment.kt:45]: I understand historical reasons now. But I'm not sure yet.

Having layout id in the constructor just will inflate it in the parent onCreateView. 

Let me research this topic more.

**Comment by v1sar** [comment]
lgtm let's merge

File: app/src/main/java/com/android/academy/fundamentals/homework/features/moviedetails/MovieDetailsFragment.kt (hunk 1 above)
Comment (emartynov) [reply app/src/main/java/com/android/academy/fundamentals/homework/features/moviedetails/MovieDetailsFragment.kt:45]: Oke, I found that when onViewCreated is called view already attached to the parent. I don't know if it matters in this situation or not.

**Pull request merged**